packages/mui-material/src/Autocomplete/Autocomplete.test.js

```diff
@@ -2418,6 +2418,26 @@ describe('<Autocomplete />', () => {
       });
       expect(textbox).to.have.property('value', 'a');
     });
+
+    it('should not throw error when nested options are provided', () => {
```

Comment:
Can you also replicate this test in the Joy UI's Autocomplete test file `packages\mui-joy\src\Autocomplete\Autocomplete.test.tsx`? 

This issue also occurs in Joy UI Autocomplete ([reproduction](https://codesandbox.io/s/mystifying-franklin-cg7s5p?file=/src/App.tsx)).

Our team is still in the process to decide whether it should be in the useAutocomplete hook test file in one place or in both the design systems (Joy UI and Material UI). But for now, we are going to replicate them.

Reply:
Sorry, I forgot about the Joy UI. Added the unit test 👍